Migrate all components to <script setup>

hanneskuettner opened the pull request:
Migrate all components to `<script setup>`.

This will be applied in individual smaller PRs, see https://github.com/directus/directus/pull/18293#issuecomment-1532102048!

The pull request remains open until we have completed the migration so that it can be kept up to date.

hanneskuettner committed:
Migrate archive-sidebar-detail

hanneskuettner committed:
Migrate revision-item

hanneskuettner committed:
Migrate accept-invite

hanneskuettner committed:
Migrate continue-as

hanneskuettner committed:
Migrate ldap-form

hanneskuettner committed:
Migrate actions

hanneskuettner committed:
Migrate shared-view

hanneskuettner committed:
Migrate render-display

hanneskuettner committed:
Migrate module-bar-avatar

hanneskuettner committed:
Migrate cards/options

hanneskuettner committed:
Migrate notification-dialogs

hanneskuettner committed:
Migrate cards/actions

hanneskuettner committed:
Migrate image-editor

hanneskuettner committed:
Migrate map/options

hanneskuettner committed:
Migrate tabular/actions

hanneskuettner committed:
Migrate displays/boolean

hanneskuettner committed:
Migrate displays/collection

hanneskuettner committed:
Migrate displays/colors (and remove unused displayTemplate)

hanneskuettner committed:
Migrate displays/file (and remove unused height)

rijkvanzanten commented:
🙈

rijkvanzanten committed:
Merge branch 'main' into feat/this-one-is-for-rijk

rijkvanzanten committed:
Update default null in cards/actions

rijkvanzanten committed:
Update layouts/cards/options

rijkvanzanten committed:
Remove default from map/actions

rijkvanzanten committed:
Tweak map/options.vue

rijkvanzanten committed:
Fix TS bug in tabular/actions

rijkvanzanten committed:
Organize imports

rijkvanzanten committed:
Organize imports

rijkvanzanten committed:
Organize imports

rijkvanzanten committed:
Remove null default from archive-sidebar-detail

rijkvanzanten committed:
Remove unnecessary default value

rijkvanzanten committed:
Organize imports

rijkvanzanten committed:
Fix nan check

rijkvanzanten committed:
Fix typing in render-display.vue

rijkvanzanten committed:
Organize imports

rijkvanzanten committed:
Use undefined instead of null

hanneskuettner committed:
Migrate private/bookmark-add

hanneskuettner committed:
Migrate private/drawer-batch

rijkvanzanten committed:
Remove default defaults

rijkvanzanten committed:
Small cleanup

rijkvanzanten committed:
🧹

hanneskuettner committed:
Migrate private/drawer-collection

Nitwel commented:
Wait, didn't somebody say we should make this incremental? 🤣

hanneskuettner committed:
Migrate private/file-lightbox

rijkvanzanten committed:
Update formatted-value

rijkvanzanten committed:
Update v-icon

rijkvanzanten committed:
Update displays/image

hanneskuettner committed:
Migrate folder-picker-list-item

rijkvanzanten committed:
Update related-values

rijkvanzanten committed:
Related-values / user

rijkvanzanten committed:
Disable eslint rule for required defaults

rijkvanzanten committed:
system-collection(s)

rijkvanzanten committed:
Improve type

rijkvanzanten committed:
System display template

rijkvanzanten committed:
system-field(-tree)

rijkvanzanten committed:
input-component

rijkvanzanten committed:
input-group

rijkvanzanten committed:
folder-list-item

hanneskuettner committed:
folder-picker

hanneskuettner committed:
header-bar-actions

rijkvanzanten committed:
folder-list-item / folder

hanneskuettner committed:
header-bar

hanneskuettner committed:
latency-indicator

hanneskuettner committed:
layout-sidebar-detail

hanneskuettner committed:
module-bar-logo

hanneskuettner committed:
module-bar

hanneskuettner committed:
notification-item

hanneskuettner committed:
notifications-drawer

hanneskuettner committed:
notifications-group

hanneskuettner committed:
notifications-preview

hanneskuettner committed:
project-info

hanneskuettner committed:
refresh-sidebar-detail

hanneskuettner committed:
revision-drawer-picker

hanneskuettner committed:
revision-drawer-preview

hanneskuettner committed:
revision-updates-change

hanneskuettner committed:
revisions-drawer-updates (changes is now always an array)

hanneskuettner committed:
revisions-drawer

hanneskuettner committed:
save-options

rijkvanzanten commented:
> Wait, didn't somebody say we should make such changes incremental? 🤣 

Haha yeaaaaaahh this is getting out of control a bit. Maybe we'll split the pr up into chunks per folder.

hanneskuettner committed:
search-input

hanneskuettner committed:
search-input (make filter optional)

hanneskuettner committed:
share-item

hanneskuettner committed:
share-sidebar-detail

hanneskuettner committed:
sidebar-button

hanneskuettner commented on app/src/views/private/components/sidebar-button.vue:
```diff
@@ -6,41 +6,35 @@
 		@click="$emit('click', $event)"
```
Unrelated to this PR, but this button will never work with `to` since it doesn't pass it on to the router-link

paescuj replied:
Will be addressed separately 👍

hanneskuettner committed:
sidebar-detail-group

hanneskuettner committed:
sidebar-detail

hanneskuettner committed:
user-popover

hanneskuettner committed:
user-invite

rijkvanzanten committed:
system-interface

rijkvanzanten committed:
system-interface-options

rijkvanzanten committed:
system-language

rijkvanzanten committed:
system-mfa-setup

rijkvanzanten committed:
system-modules

rijkvanzanten committed:
system-scope

rijkvanzanten committed:
boolean

rijkvanzanten committed:
datetime

rijkvanzanten committed:
accordion-section

rijkvanzanten committed:
group-accordion

paescuj commented on .eslintrc.js:
```diff
@@ -81,6 +81,7 @@ module.exports = {
 			rules: {
 				...defaultRules,
 				'vue/multi-word-component-names': 'off',
+				'vue/require-default-prop': 'off',
```
Generally, I'm in favor of this. Just wondering if there are any downsides that might come with this as it is a strongly recommended rule: https://eslint.vuejs.org/rules/require-default-prop.html

rijkvanzanten replied:
Hannes and I had the same doubt. I think typescript makes this rule redundant. The goal of the rule is to make sure you can reliably use a prop with the expected type even if it's omitted (so you don't accidentally try to use undefined). Typescript already handles that type safety!

paescuj replied:
I think you're right 😃👍

paescuj committed:
Use same ordering of script props everywhere

- Following official docs: https://vuejs.org/guide/typescript/composition-api.html#typing-component-props
- Same ordering so it's easily "grep-able"

paescuj committed:
Cleanup & same definition of 'inheritAttrs' everywhere

- Move to top so it's well visible
- 'defineComponent' not necessary - wrong type detected by volar / vue-tsc anyway

paescuj committed:
Remove explicit component name definitions

No longer necessary, see https://vuejs.org/api/options-misc.html#name

paescuj committed:
Remove unused vars in views/private/components/share-item

paescuj committed:
routes/login/components/login-form/login-form

paescuj committed:
routes/login/components/sso-links

paescuj committed:
routes/logout

paescuj committed:
routes/private-not-found

paescuj committed:
routes/reset-password/request

paescuj committed:
routes/reset-password/reset

paescuj committed:
routes/reset-password/reset-password

paescuj committed:
routes/shared/components/share-item

paescuj committed:
routes/shared/shared

paescuj committed:
routes/tfa-setup

paescuj committed:
modules/users/routes/item

paescuj committed:
modules/users/components/user-info-sidebar-detail

paescuj committed:
modules/users/components/navigation

paescuj committed:
modules/users/components/navigation-role

paescuj committed:
modules/settings/routes/webhooks/item

paescuj committed:
modules/settings/routes/webhooks/collection

paescuj committed:
modules/settings/routes/roles/public-item

paescuj committed:
modules/settings/routes/roles/permissions-detail/permissions-detail

paescuj committed:
modules/settings/routes/roles/permissions-detail/components/validation

paescuj committed:
modules/settings/routes/roles/permissions-detail/components/tabs

paescuj committed:
modules/settings/routes/roles/permissions-detail/components/presets

paescuj committed:
modules/settings/routes/roles/permissions-detail/components/permissions

paescuj committed:
modules/settings/routes/roles/permissions-detail/components/fields

paescuj committed:
modules/settings/routes/roles/permissions-detail/components/actions

paescuj committed:
modules/settings/routes/roles/item/item

paescuj committed:
modules/settings/routes/roles/collection

paescuj committed:
modules/settings/routes/roles/add-new

paescuj committed:
modules/settings/routes/project/project

paescuj committed:
modules/settings/routes/project/components/project-info-sidebar-detail

paescuj committed:
modules/settings/routes/presets/collection/components/presets-info-sidebar-detail

paescuj committed:
modules/settings/routes/presets/collection/collection

paescuj committed:
modules/settings/routes/not-found

paescuj committed:
modules/settings/routes/data-model/new-collection

paescuj committed:
modules/settings/routes/data-model/fields/fields

paescuj committed:
modules/settings/routes/data-model/fields/components/fields-management

paescuj committed:
modules/activity/components/navigation

paescuj committed:
modules/activity/routes/collection

paescuj committed:
modules/activity/routes/item

paescuj committed:
modules/content/components/navigation-item-content

paescuj committed:
modules/content/components/navigation-item

paescuj committed:
modules/content/components/navigation

paescuj committed:
modules/content/routes/collection-or-item

paescuj committed:
modules/content/routes/collection

paescuj committed:
modules/content/routes/no-collections

paescuj committed:
modules/content/routes/not-found

paescuj committed:
modules/files/components/add-folder

paescuj committed:
modules/files/components/file-info-sidebar-detail

paescuj committed:
modules/files/routes/not-found

paescuj committed:
modules/insights/routes/not-found

paescuj committed:
modules/files/components/navigation-folder

paescuj committed:
modules/files/components/navigation

paescuj committed:
modules/files/routes/add-new

paescuj committed:
modules/files/routes/collection

paescuj committed:
modules/insights/components/dashboard-dialog

paescuj committed:
modules/insights/components/navigation

paescuj committed:
modules/insights/routes/overview

paescuj committed:
modules/settings/components/navigation

paescuj committed:
modules/settings/routes/data-model/collections/collections

paescuj committed:
modules/settings/routes/data-model/collections/components/collection-dialog

paescuj committed:
modules/settings/routes/data-model/collections/components/collection-item

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-actions

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-display

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-field

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-interface

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship-m2a

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship-m2m

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship-o2m

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship-translations

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-schema

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-tabs

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-simple/field-configuration

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-simple/relationship-configuration

paescuj committed:
modules/settings/routes/data-model/field-detail/shared/extension-options

paescuj committed:
modules/settings/routes/data-model/field-detail/shared/related-collection-select

paescuj committed:
modules/settings/routes/data-model/field-detail/shared/related-field-select

paescuj committed:
modules/settings/routes/data-model/fields/components/field-select-menu

paescuj committed:
modules/settings/routes/data-model/fields/components/field-select

paescuj committed:
modules/settings/routes/roles/item/components/permissions-overview-row

paescuj committed:
modules/settings/routes/roles/item/components/permissions-overview

paescuj committed:
modules/settings/routes/roles/item/components/permissions-overview-toggle

paescuj committed:
modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship-m2o

paescuj committed:
interfaces/presentation-divider/presentation-divider

paescuj committed:
interfaces/group-detail/group-detail

paescuj committed:
interfaces/input-rich-text-html/input-rich-text-html

paescuj committed:
interfaces/input/input

paescuj committed:
interfaces/group-raw/group-raw

paescuj committed:
interfaces/input-rich-text-md/input-rich-text-md

paescuj committed:
interfaces/slider/slider

paescuj committed:
interfaces/presentation-notice/presentation-notice

paescuj committed:
interfaces/select-dropdown/select-dropdown

paescuj commented on app/src/displays/icon/icon.vue:
```diff
@@ -2,33 +2,23 @@
+		filled?: boolean;
+	}>(),
+	{
+		color: undefined,
```
I think defaults with `undefined` can be omitted. Same in other files.

paescuj committed:
Remove unnecessary boolean defaults

paescuj committed:
interfaces/map/map

paescuj committed:
interfaces/map/options

paescuj committed:
interfaces/select-multiple-checkbox-tree/select-multiple-checkbox-tree

paescuj committed:
interfaces/select-icon/select-icon

paescuj committed:
interfaces/translations/language-select

paescuj committed:
layouts/calendar/actions

paescuj committed:
layouts/calendar/options

paescuj committed:
interfaces/tags/tags

paescuj committed:
interfaces/list/list

paescuj committed:
interfaces/list/options

paescuj committed:
interfaces/input-code/input-code

paescuj committed:
interfaces/select-multiple-checkbox/select-multiple-checkbox

paescuj committed:
interfaces/input-multiline/input-multiline

paescuj committed:
interfaces/select-multiple-dropdown/select-multiple-dropdown

paescuj committed:
interfaces/select-radio/select-radio

paescuj committed:
interfaces/input-autocomplete-api/input-autocomplete-api

paescuj committed:
interfaces/input-hash/input-hash

paescuj committed:
layouts/cards/components/card

paescuj committed:
Fix optional prop in modules/files/components/navigation-folder

paescuj committed:
layouts/cards/components/header

paescuj committed:
layouts/map/components/map

paescuj committed:
layouts/map/map

paescuj committed:
layouts/cards/cards

paescuj committed:
Merge branch 'main' into feat/this-one-is-for-rijk

rijkvanzanten committed:
Merge branch 'main' into feat/this-one-is-for-rijk

hanneskuettner committed:
Fix css typo in interfaces/list/list.vue

hanneskuettner committed:
Add v-if to list in interfaces/translations/language-select.vue for proper types

hanneskuettner commented on app/src/layouts/calendar/actions.vue:
```diff
@@ -7,21 +7,16 @@
-import { defineComponent } from 'vue';
-
-export default defineComponent({
+export default {
```
`export default defineComponent({`
Let's be good about this. Same goes for all other instances of this.

paescuj replied:
See https://github.com/hanneskuettner/directus/commit/68ae14da5876d9bb7714f2f40e6e85785f37a8fa.

We'll align those exports in a separate PR.

hanneskuettner committed:
Fix required props in layouts/cards/options.vue

hanneskuettner commented on app/src/modules/files/components/navigation-folder.vue:
```diff
@@ -113,180 +113,156 @@
+	defineProps<{
+		folder: Folder;
+		currentFolder?: string;
+		clickHandler?: () => void;
```
`clickHandler` seems to be completely unused (only passed down to a nested `navigation-folder` and never called. Also it is never bound in the one instance that this component is used. Pretty sure it is safe to remove.

paescuj replied:
Is reworked in https://github.com/directus/directus/pull/18320 anyway 👍

hanneskuettner commented on app/src/modules/insights/components/dashboard-dialog.vue:
```diff
@@ -29,83 +29,77 @@
+		dashboard?: Dashboard;
+	}>(),
+	{
+		modelValue: false,
```
- Remove unnecessary `modelValue: false`. Booleans default to `false` automatically. Or do we wan't to be explicit here since it is a modelValue and not a simple flag?

hanneskuettner commented on app/src/modules/settings/routes/data-model/collections/components/collection-item.vue:
```diff
@@ -66,108 +66,90 @@ import { DeepPartial } from '@directus/types';
+		disableDrag?: boolean;
+	}>(),
+	{
+		disableDrag: false,
```
Remove explicit default.

paescuj commented on app/src/displays/formatted-value/formatted-value.vue:
```diff
@@ -19,196 +19,161 @@
+		type: string;
+		value?: string | number | (string | number)[];
+		format?: boolean;
+		font?: string;
```
```suggestion
		font?: 'sans-serif' | 'serif' | 'monospace';
```

paescuj committed:
Fix font prop type

paescuj commented on app/src/displays/labels/labels.vue:
```diff
@@ -33,73 +33,58 @@ type Choice = {
+const props = withDefaults(
+	defineProps<{
+		value: string | string[];
+		type: string;
```
Thanks!

paescuj commented on app/src/displays/labels/labels.vue:
```diff
@@ -33,73 +33,58 @@ type Choice = {
+const props = withDefaults(
+	defineProps<{
+		value: string | string[];
+		type: string;
```
```suggestion
		type: 'text' | 'string' | 'json' | 'csv';
```

paescuj committed:
Fix "type" prop type

rijkvanzanten commented:
There's gonna be no way to merge this as one giant PR. Lets close this one out in favor of individual smaller PRs we can actually review and merge 😄 

I'll update this comment with all other PRs that are related:

- #18388

paescuj committed:
Merge branch 'main' into feat/this-one-is-for-rijk

Based on commit "Require curly brackets in if statements when they're multi-line
(#18387)"

paescuj committed:
Merge branch 'main' into feat/this-one-is-for-rijk

paescuj committed:
Revert "Cleanup & same definition of 'inheritAttrs' everywhere"

This reverts commit 68ae14da5876d9bb7714f2f40e6e85785f37a8fa.

Let's just not forget to apply this later on... 🙏

paescuj committed:
Revert "Remove explicit component name definitions"

This reverts commit 5e63e83615394808a9781a7c9bbca48d40383108.

Let's just not forget to apply this later on... 🙏

paescuj committed:
Revert "Use same ordering of script props everywhere"

This reverts commit b3dae0eaca3308787a971e093620c4d2d52c3009.

Let's just not forget to apply this later on... 🙏

paescuj committed:
Sync with already merged components

paescuj committed:
Revert "unrelated" changes in v-image

We probably want to do such optimizations for all components anyway...

paescuj committed:
Merge branch 'main' into feat/this-one-is-for-rijk

paescuj committed:
Improve interfaces/list

paescuj committed:
Clean-up notifications-drawer

paescuj committed:
Fix interfaces/map

paescuj commented on app/src/interfaces/presentation-notice/presentation-notice.vue:
```diff
@@ -6,25 +6,19 @@
+	{
+		color: 'normal',
+		icon: 'info',
+		text: 'No text configured...',
```
Thanks, will be addressed separately 👍

paescuj commented on app/src/interfaces/translations/language-select.vue:
```diff
@@ -28,34 +28,25 @@
+const props = withDefaults(
+	defineProps<{
+		modelValue?: string;
+		items?: Record<string, any>[];
```
Will be properly typed later on 👍

paescuj committed:
Improve activity/routes/item

paescuj committed:
Improve interfaces/list again

paescuj committed:
Finish interfaces/list/options

paescuj commented on app/src/modules/files/routes/collection.vue:
```diff
@@ -219,458 +219,402 @@ type Item = {
+const props = defineProps<{
+	folder?: string;
+	special?: 'all' | 'recent' | 'mine';
+}>();
```
Addressed in #18320 👍

paescuj committed:
Remove unnecessary default value in dashboard-dialog

paescuj committed:
Type defineEmits in dashboard-dialog

paescuj committed:
Remove unnecessary default value in collection-item

paescuj committed:
Merge branch 'main' into feat/this-one-is-for-rijk

paescuj committed:
Revert unrelated changes

paescuj committed:
Remove unnecessary default value in collection-dialog

paescuj committed:
Add check back in field-detail-advanced-relationship-translations

Not sure about this one, better safe than sorry

paescuj committed:
Remove unnecessary default value in extension-options

paescuj committed:
Remove unnecessary default value in related-collection-select

paescuj committed:
Remove unnecessary default value in related-field-select

paescuj committed:
Remove unnecessary default value in field-select-menu

rijkvanzanten commented:
Aaaand we're done! Thanks @hanneskuettner, @paescuj! ❤️ 🚀
